src/string.cr

```diff
@@ -563,13 +563,18 @@ class String
     !!index(str)
   end
 
-  def split
+  def split(limit = nil : Int32?)
+    if limit && limit <= 1
```

Comment:
Both do the same thing, it's a matter of style. I'm fine with both. Sometimes I find one way more readable than the other, but I'm not sure I know what's the algorithm in my head for that :-P